cdc/owner/gc_manager.go

```diff
@@ -122,8 +126,15 @@ func (m *gcManager) CheckStaleCheckpointTs(ctx cdcContext.Context, checkpointTs
 	if err != nil {
 		return errors.Trace(err)
 	}
-	if checkpointTs < m.lastSafePointTs || pdTime.Sub(oracle.GetTimeFromTS(checkpointTs)) > time.Duration(m.gcTTL)*time.Second {
-		return cerror.ErrSnapshotLostByGC.GenWithStackByArgs(checkpointTs, m.lastSafePointTs)
+	if m.isTiCDCBlockGC {
+		if pdTime.Sub(oracle.GetTimeFromTS(checkpointTs)) > time.Duration(m.gcTTL)*time.Second {
+			return cerror.ErrSnapshotLostByGC.GenWithStackByArgs(checkpointTs, m.lastSafePointTs)
```

Comment:
Probably using the error `ErrSnapshotLostByGC` not appropriate here, as it may mislead the user into thinking that some data has actually been lost.